src/lib/entity.js

```diff
@@ -602,8 +602,9 @@ export function printEntity(entity, onDoubleClick) {
  * @return {Element} Entity created
  */
 export function createEntity(definition, cb) {
+  const re = /^[\'a-zA-Z0-9 _-]{1,50}$/;
   let objName = "";
-  while(objName == ""){
+  while(objName == "" || !re.test(objName)){
```

Comment:
Instead of putting it like this, let's change the order so we can tell the user what went wrong, something like:
```
while(objName == ""){
objName = prompt("Enter the name of the object");
if(!re.test(objName)){
  alert("Please enter a name with only alphanumeric, space, underscore or dash characters only")
  objName = "";
}
}

```